RatFinder/app/src/main/java/team45/ratfinder/controller/StartActivity.java

```diff
@@ -72,14 +70,14 @@ public void onCancelled(DatabaseError databaseError) {
         //DO NOT DELETE THIS LINE OF CODE:
         //This query will find the first 50 rat sightings and the recycler view will be filled with
         //these. Later, there will be queries based on location, date, etc.
-        Query sightingsListQuery = sightingsListReference.orderByChild("Latitude").limitToFirst(50);
+        Query sightingsListQuery = sightingsListReference.orderByChild("Created Date").limitToLast(30); //currently sorting by date
         sightingsListQuery.addChildEventListener(new ChildEventListener() {
             @Override
             public void onChildAdded(DataSnapshot dataSnapshot, String s) {
-                //Log.d("Firebase", dataSnapshot.getValue().toString());
+                Log.d("Firebase", dataSnapshot.getValue().toString());
                 RatSighting ratSighting = FirebaseObjectConverter
                         .getRatSighting((Map)dataSnapshot.getValue(), dataSnapshot.getKey());
-                sightingsList.add(ratSighting);
+                sightingsList.add(0, ratSighting);
                 ratSightingRecyclerViewAdapter.notifyDataSetChanged();
```

Comment:
The reason it takes the home page so long to load is because you add elements to the front of the ArrayList. This means that all of the elements already in the array have to shift every time, making add an O(n) operation instead of O(1). This means that adding n items to the array is O(n^2) instead of O(n). The quickest solution is to add elements to the back, or just make it a LinkedList instead of an ArrayList

Reply:
i'm adding to the front so that the most recent entries are shown first.
the slowness is not due to this for two reasons:
1. if you add to the back, it will still be just as slow(i tried)
2. the built in java arraylist has efficiency countermeasures if you're adding to the front by default so the real big o is closer to O(n) then to n^2.
3. the slowdown comes from firebase shifting its 'viewing area' to the most recent sightings and then sorting the longs.

Reply:
Thanks for looking into this